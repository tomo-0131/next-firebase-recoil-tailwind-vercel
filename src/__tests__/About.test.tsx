/**
 * @jest-environment jsdom
 */
import { render } from "@testing-library/react";
import About from "../pages/about";

describe('about', () => {
  // let container: HTMLDivElement
  // beforeEach(() => {
  //   document.body.appendChild(container = document.createElement('div'))
  // })
  // afterEach(() => {
  //   document.body.removeChild(container)
  // })

  it('should title render', () => {
    const container = render(<About />)
    expect(container.getByText('⚡️Talexy')).toMatchSnapshot()
  })

  it ('img must have val = env.local', () => {
    const container = render(<About />);
    expect(container.getByRole('img')).toMatchSnapshot();
  })
})
