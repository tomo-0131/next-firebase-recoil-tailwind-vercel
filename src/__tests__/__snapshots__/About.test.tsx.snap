// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`about img must have src = env.test 1`] = `
<img
  alt="AboutHeroset"
  class="flex justify-center items-center my-1 pb-12 md:mx-auto lg:mx-auto"
/>
`;

exports[`about img must have val = env.local 1`] = `
<img
  alt="AboutHero"
  class="my-1 pb-12 md:mx-auto lg:mx-auto"
  data-nimg="intrinsic"
  decoding="async"
  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover;"
/>
`;

exports[`about img must have val = env.test.local 1`] = `
<img
  alt="AboutHero"
  class="my-1 pb-12 md:mx-auto lg:mx-auto"
  data-nimg="intrinsic"
  decoding="async"
  src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
  style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: cover;"
/>
`;

exports[`about should stay calm 1`] = `
Array [
  <h1
    class="text-2xl my-3 p-5 text-center lg:text-3xl lg:p-9 lg:my-3"
  >
    ⚡️Talexy
  </h1>,
]
`;

exports[`about should title render 1`] = `
<h1
  class="text-2xl my-3 p-5 text-center lg:text-3xl lg:p-9 lg:my-3"
>
  ⚡️Talexy
</h1>
`;
